code/game/objects/effects/decals/cleanable.dm

```diff
@@ -84,6 +87,40 @@ var/list/infected_cleanables = list()
 				if (L.client)
 					L.client.images |= pathogen
 
+
+/obj/effect/decal/cleanable/proc/fixDNA()
+	if (!(blood_DNA?.len))
+		blood_DNA[fake_DNA] = "N/A"
```

Comment:
this will runtime if `blood_DNA` is `null`, which is apparently a condition you're expecting, according to your `if`

Reply:
This proc is only called in New() where blood_DNA is list() by default, but yeah that's a fair assessment.